### DISCUSSION TITLE
Regex and Polish letters in Atom find in buffer panel

### DISCUSSION THREAD

**Discussion by Andrzej-Andrzej** [atom/atom#23508] [Q&A]
Hi, 
I would like to match these words: "Łucja", "Dąbrówka", "Ścibor", "Miłosz".
In general I would like to find all Polish diacritic characters in Atom for other words containing Polish letters as well.
I tried this: "^[A-ZŁŚ]{1}+[a-ząęółśżźćń]+$"    but it doesn't work.
Any help will be greatly appreciated.
Thank you

**Comment by icecream17** [comment]
You can check regex101.com or similar (search regex checker)

In this case:

```^[A-ZŁŚ]{1}+[a-ząęółśżźćń]+$``` old\
```^[A-ZŁŚ]{1}[a-ząęółśżźćń]+$``` new

Also I don't think the start/end marks are necessary (^$)



Although... I think this would match all words, even ones without diacritics (example: "Abcdefz")

This probably isn't your goal.

So I think it should go like:

(\
[Capital Diacritic]{1}[English or diacritic]+\
or\
[Capital English]{1}[English]\*[diacritic]{1}[English or diacritic]*\
)

So

```[ŁŚ]{1}[a-ząęółśżźćń]+|[A-Z]{1}[a-z]*[ąęółśżźćń]{1}[a-ząęółśżźćń]*```
